stainless_extraction/src/literal.rs

```diff
@@ -30,36 +29,30 @@ impl Literal {
       Literal::String(value) => f.StringLiteral(value.clone()).into(),
     }
   }
-}
-
-impl<'tcx> TryFrom<&'tcx ty::Const<'tcx>> for Literal {
-  type Error = ();
 
-  fn try_from(konst: &'tcx ty::Const<'tcx>) -> Result<Self, Self::Error> {
+  pub fn from(konst: &ty::Const<'_>, tcx: TyCtxt<'_>) -> Option<Self> {
```

Comment:
I would in general reserve the `from` name for `From` instances.

```suggestion
  pub fn from_const(konst: &ty::Const<'_>, tcx: TyCtxt<'_>) -> Option<Self> {
```